connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/WorkerErrantRecordReporter.java

```diff
@@ -49,7 +54,7 @@
     private final HeaderConverter headerConverter;
 
     // Visible for testing
-    protected final LinkedList<Future<Void>> futures;
+    protected final Map<TopicPartition, Future<Void>> futures;
```

Comment:
If we want this to be a concurrent map probably good idea to depict this in the declaration. 
(same as you do in the tests below)

```suggestion
    protected final ConcurrentMap<TopicPartition, Future<Void>> futures;
```

Reply:
I like this style and agree with the suggestion; however, there's actually a bug in the PR right now where we only store one `Future` at a time per topic-partition, so will address in a follow-up commit instead of just clicking the `Commit suggestion` button.